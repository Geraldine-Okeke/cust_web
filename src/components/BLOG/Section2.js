import React, { useState, useEffect } from 'react';
async function fetchVid() {
  const API_KEY = 'AIzaSyAMCyd-bir3PHApWkH8GXQcZeJQ1SY6MDo';
  const CHANNEL_ID = 'UC-GAShuXH58oh9GoaJ-f6Sg';

  const response = await fetch(
    `https://www.googleapis.com/youtube/v3/search?key=${API_KEY}&channelId=${CHANNEL_ID}&part=snippet&type=video`
  );

  const data = await response.json();
  return data.items; // This array contains video information
}
export default function Section2() {
  const [videos, setVideos] = useState([]);

  useEffect(() => {
    async function loadVideos() {
      const videoItems = await fetchVid();
      setVideos(videoItems);
    }

    loadVideos();
  }, []);

  return (
    <div>
      <h2 className='text-3xl font-bold p-3' >YouTube Videos</h2>
      <div className="grid grid-cols-1 md:grid-cols-2 gap-4">
        {videos.map(video => (
          <div key={video.id.videoId} className="p-4">
            <a
              href={`https://www.youtube.com/watch?v=${video.id.videoId}`}
              target="_blank"
              rel="noopener noreferrer"
            >
              <div className="relative aspect-w-16 aspect-h-9">
                <iframe
                  className="w-full h-full"
                  src={`https://www.youtube.com/embed/${video.id.videoId}`}
                  title={video.snippet.title}
                  frameBorder="0"
                  allowFullScreen
                />
              </div>
              <p>{video.snippet.title}</p>
            </a>
            <p>{video.snippet.description}</p>
          </div>
        ))}
      </div>
    </div>
  );
}
