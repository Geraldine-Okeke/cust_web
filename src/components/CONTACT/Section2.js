import { Link } from 'react-router-dom'
import { FaFacebook, FaTwitter, FaInstagram } from 'react-icons/fa';
const Section2 =()=>{
  return (
    <div className=' contact h-auto text-white rounded-b-xl'>
      <h1 className="text-3xl font-extrabold text-center mt-10">CONTACT INFORMATION</h1>
      <div className="flex flex-wrap mt-10 pb-20">
        <div className="w-full md:w-1/3 p-4">
          <h2 className="text-2xl font-bold">Address</h2>
          <hr className='my-2 border-t-2 border-yellow-500 w-1/4' />
          <p className='leading-10'>The Grey Gate, Olushola Agbaje, Off Road 13 lekki Phase I, Lagos, Nigeria.</p>
        </div>
        <div className="w-full md:w-1/3 p-4  ">
          <h2 className="text-2xl font-bold md:text-center">Phone and Email</h2>
          <hr className='my-2 border-t-2 border-yellow-500 w-1/4 md:mx-auto'/>
          <p className='md:text-center pb-7'> +234 818 038 0006</p>
          <p className='md:text-center'>info@kindlegatefoundation.org</p>
        </div>
        <div className="w-full md:w-1/3 p-4 h-auto">
          <h2 className="text-2xl font-bold md:text-center">Social Media</h2>
          <hr className='my-2 border-t-2 border-yellow-500 w-1/4 md:mx-auto' />
          <div className="flex md:justify-center items-center ">
            <div className="w-auto p-2">
              <Link to="https://www.facebook.com/kindlegatefoundation">
                <FaFacebook size={22} />
              </Link>
            </div>
            <div className="w-auto p-2">
              <Link to="https://twitter.com/kindlegatecares">
                <FaTwitter size={22} />
              </Link>
            </div>
            <div className="w-auto p-2">
              <Link to="https://www.instagram.com/kindlegatefoundation/?hl=en">
                <FaInstagram size={22} />
              </Link>
            </div>
          </div>
          
        </div>
      </div>
     
    </div>
    
  );

}
export default Section2