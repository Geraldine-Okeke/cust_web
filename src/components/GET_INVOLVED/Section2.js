export default function Section2(){
  return(
    <>
      <div className="md:w-3/4 mx-auto md:text-center py-20">
      <h1 className="text-3xl font-bold pb-10 ">VOLUNTEER, MENTOR OR SPONSOR</h1>
        <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor 
          incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, 
          quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor
        </p>
        <p className="pt-10">Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor 
          incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud 
          exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor
        </p>
      </div>
    </>
  )
}