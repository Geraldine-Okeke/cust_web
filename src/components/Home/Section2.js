import Sec2_img from './Images/Section2.png'
import { Link } from 'react-router-dom';
export default function Section2(){
  return(
    <>
    <section>
      <div className='grid grid-cols-2 gap-4 bg-gray-200 pb-10 mb-10 rounded-xl '>
        <div className='mt-10'>
          <img className='-ml-20 mt-3' src={Sec2_img} alt=""/>
        </div>
        <div className='mt-10'>
          <div className='mt-5 mb-10'>
            <div className='flex flex-col items-end'>
              <h2 className='text-2xl font-bold '>OUR MANDATE</h2>
              <hr className='float-left w-20 font-extrabold border-t-2 bg-custom-yellow mb-4' />
            </div>
            <div>
              <p className=''>
              Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Maecenas porttitor congue massa. 
              Fusce posuere, magna sed pulvinar ultricies, purus lectus malesuada libero, sit amet commodo
              magna eros quis urna. Nunc viverra imperdiet enim. Fusce est. Vivamus a tellus. Pellentesque 
              habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. 
              </p>
            </div>
            
          </div>
          <div className='flex flex-col'>
            <div className=' flex flex-col items-end'>
              <h2 className='text-2xl font-bold'>OBJECTIVES</h2>
              <hr className=' w-20 font-bold border-t-2 bg-custom-yellow mb-4' />
            </div>
            <div>
              <p className=''>
              Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Maecenas porttitor congue 
              massa. Fusce posuere, magna sed pulvinar ultricies, purus lectus malesuada libero, 
              sit amet commodo magna eros quis urna. Nunc viverra imperdiet enim. Fusce est. Vivamus 
              a tellus. Pellentesque habitant morbi tristique senectus et netus et malesuada fames 
              ac turpis egestas. 
              </p>
            </div>
            
          </div>
          <div className='py-10 sm:float-right mr-0 sm:mr-20'>
            <Link to='/MANDATE/Mandate'>
              <button className=' bg-blue-900 hover:bg-blue-600 text-black font-semibold  whitespace-nowrap h px-7 rounded-full padding  w-auto h-14' >LEARN MORE</button>
            </Link>
            
          </div>
           
        </div>
        
      </div>
    </section>
    
    </>
  )
}