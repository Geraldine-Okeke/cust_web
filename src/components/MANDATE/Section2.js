import image from '../Home/Images/Section2.png'
import './Mandate.css'
import React, { useEffect, useRef } from 'react';
export default function Section2(){
  const aboutSectionRef = useRef(null); 
  useEffect(() => {
    const locationHash = window.location.hash;
    if (locationHash === '#about') {
      aboutSectionRef.current.scrollIntoView({ behavior: 'smooth' });
    }
  }, []);
  return (
    <>
    <div className="">
      <div className="grid grid-cols-3 gap-3.5">
        <div className="col-span-3 p-4 md:col-span-2 md:">
          <div className="w-full py-10 md:w-3/4 md:p-10" ref={aboutSectionRef} id='about'>
            <h1 className="font-extrabold text-2xl">About Us</h1>
            <p>
            Lorem ipsum dolor sit amet, consectetuer adipiscing elit. 
            Maecenas porttitor congue massa. Fusce posuere, magna sed pulvinar ultricies,
            </p>
            <br/>
            <p>
            purus lectus malesuada libero, sit amet commodo magna eros quis urna. Nunc viverra imperdiet enim. Fusce est. Vivamus a tellus. Pellentesque habitant
            </p>
            <br/>
            <p>
            morbi tristique senectus et netus et malesuada fames ac turpis egestas. Lorem ipsum
            dolor sit amet, consectetuer adipiscing Maecenasporttitor congue massa. Fusce posuere, magna sed pulvinar ultricies,
            puruslectus malesuada libero, sit amet commodo magna eros quis urna. Nunc viverra
            </p>
            
          </div>
          <div className="w-full py-10 md:w-3/4 md:p-10">
            <h1 className="font-extrabold text-2xl">Our History</h1>
            <p>
            Lorem ipsum dolor sit amet, consectetuer adipiscing elit.Maecenas porttitor congue massa. Fusce posuere, magna sed pulvinar ultricies,
            </p>
            <br/>
            <p>
            purus lectus malesuada libero, sit amet commodo magna eros quis urna. Nunc viverra imperdiet enim. Fusce est. Vivamus a tellus. Pellentesque habitant
            </p>
            <br/>
            <p>
            morbi tristique senectus et netus et malesuada fames ac turpis egestas. Lorem ipsum
            dolor sit amet, consectetuer adipiscing Maecenasporttitor congue massa. Fusce posuere, magna sed pulvinar ultricies,
            puruslectus malesuada libero, sit amet commodo magna eros quis urna. Nunc viverra
            </p>
            
          </div>
        </div>
        <div className="hidden md:block p-4 ml-1/4 md:w-full relative w-1/2">
          <img className='absolute -right-32  top-48 round' src={image} alt=''/>
        </div>
      </div>
    </div>
    </>
  )
}
