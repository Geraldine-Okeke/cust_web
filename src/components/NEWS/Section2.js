import React from 'react';
import Slide from './Slide';
export default function Section2() {
  return (
    <div className="yellow news rounded-b-xl ">
      <h1 className='text-center font-bold text-4xl p-12'>News</h1>
      <div className="flex px-3 flex-col md:flex-row">
        <div className="md:w-1/2  w-auto">
          <span className="font-bold text-black md:p-4">What is happening now</span>
          <h1 className="text-white font-bold text-2xl md:text-4xl md:p-4 w-auto ">GATE FOUNDATION IN COLLABORATION WITH ...</h1>
        </div>
        <div className="md:w-1/2 ">
          <p className='md:p-4'>
            “Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et
            dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip
            ex ea commodo consequat. Duis aute irure dolor”
          </p>
        </div>
      </div>

      <div className="flex justify-center  p-6   ">
        <Slide />
      </div>

    </div>
  );
}
