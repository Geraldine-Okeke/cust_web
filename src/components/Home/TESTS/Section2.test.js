import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom'; 
import Section2 from '../Section2'; 

test('Section2 component snapshot', () => {
  const component = renderer.create(
    <MemoryRouter>
      <Section2 />
    </MemoryRouter>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
