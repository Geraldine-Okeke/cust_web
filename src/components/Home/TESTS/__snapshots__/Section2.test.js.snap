// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section2 component snapshot 1`] = `
<section>
  <div
    className="grid grid-cols-2 gap-4 bg-gray-200 pb-10 mb-10 rounded-xl "
  >
    <div
      className="mt-10"
    >
      <img
        alt=""
        className="-ml-20 mt-3"
        src="Section2.png"
      />
    </div>
    <div
      className="mt-10"
    >
      <div
        className="mt-5 mb-10"
      >
        <div
          className="flex flex-col items-end"
        >
          <h2
            className="text-2xl font-bold "
          >
            OUR MANDATE
          </h2>
          <hr
            className="float-left w-20 font-extrabold border-t-2 bg-custom-yellow mb-4"
          />
        </div>
        <div>
          <p
            className=""
          >
            Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Maecenas porttitor congue massa. Fusce posuere, magna sed pulvinar ultricies, purus lectus malesuada libero, sit amet commodo magna eros quis urna. Nunc viverra imperdiet enim. Fusce est. Vivamus a tellus. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas.
          </p>
        </div>
      </div>
      <div
        className="flex flex-col"
      >
        <div
          className=" flex flex-col items-end"
        >
          <h2
            className="text-2xl font-bold"
          >
            OBJECTIVES
          </h2>
          <hr
            className=" w-20 font-bold border-t-2 bg-custom-yellow mb-4"
          />
        </div>
        <div>
          <p
            className=""
          >
            Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Maecenas porttitor congue massa. Fusce posuere, magna sed pulvinar ultricies, purus lectus malesuada libero, sit amet commodo magna eros quis urna. Nunc viverra imperdiet enim. Fusce est. Vivamus a tellus. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas.
          </p>
        </div>
      </div>
      <div
        className="py-10 sm:float-right mr-0 sm:mr-20"
      >
        <a
          href="/MANDATE/Mandate"
          onClick={[Function]}
        >
          <button
            className=" light-blue hover:bg-blue-600 text-black font-semibold  h px-2 rounded-full sm:w-56 w-28 h-14"
          >
            LEARN MORE
          </button>
        </a>
      </div>
    </div>
  </div>
</section>
`;
